src/nodes.coffee

```diff
@@ -4664,6 +4668,25 @@ exports.For = class For extends While
     fragments.push @makeCode(returnResult) if returnResult
     fragments
 
+  astType: -> 'For'
```

Comment:
I chose to make `for` represented as a new (non-Babel) AST type `For`

For context, the possible corresponding Babel AST types would be `ForStatement`, `ForInStatement` and `ForOfStatement`

Our `for` constructs don't correspond to the parts expected by `ForStatement` (eg `for (i = 0; i < n; i++) {...}`), so that's not an option

In theory we could try and use `ForInStatement` for `for ... of` and `ForOfStatement` for `for ... from` since those correlate. But:
- we'd still have other `for` constructs (`for ... in` and `for <range>`) that we'd need to make up new types for
- it's confusing that the names don't match up, ie `for ... of` isn't `ForOfStatement`, it's `ForInStatement`

All of the different `for` constructs have similar "extension" children/properties (eg guard, step, postfix) so to me it made sense to unify them as a single AST node type and then indicate which construct (ie `for ... in`, `for ... of`, `for ... from`, or `for <range>`) it is via an AST property (specifically `style`, which can be `in`, `of`, `from` or `range`)

So that's why I went with a unique `For` AST node type that encompasses all of them

Reply:
Well it sounds like Babel's version is a lower level of ours, like what ours transpiles into. Not sure we have much of a choice if we want to preserve the original content.